### PR TITLE
Finish Typescripterizing the frontend

### PR DESCRIPTION
**Pull request by paulmelnikow**

### CODE REVIEW

File: frontend/components/badge-examples.tsx [hunk 1]
```diff
@@ -7,23 +7,13 @@ import {
 } from '../../core/badge-urls/make-badge-url'
 import { removeRegexpFromPattern } from '../lib/pattern-helpers'
 import {
-  ExampleSignature,
   Example as ExampleData,
+  Suggestion,
+  RenderableExample,
 } from '../lib/service-definitions'
 import { Badge } from './common'
 import { StyledCode } from './snippet'
```
Comment (paulmelnikow) [inline frontend/components/badge-examples.tsx]: These were moved to `frontend/lib/service-definitions/index.ts`.

File: frontend/components/badge-examples.tsx [hunk 2]
```diff
@@ -49,19 +39,19 @@ function Example({
   isBadgeSuggestion,
 }: {
   baseUrl?: string
-  onClick: (exampleData: RenderableExampleData) => void
-  exampleData: RenderableExampleData
+  onClick: (example: RenderableExample, isSugestion: boolean) => void
```
Comment (paulmelnikow) [inline frontend/components/badge-examples.tsx]:
```suggestion
  onClick: (example: RenderableExample, isSuggestion: boolean) => void
```

File: frontend/components/badge-examples.tsx [hunk 3]
```diff
@@ -112,10 +102,10 @@ export function BadgeExamples({
   baseUrl,
   onClick,
 }: {
-  examples: RenderableExampleData[]
+  examples: RenderableExample[]
   areBadgeSuggestions: boolean
   baseUrl?: string
-  onClick: (exampleData: RenderableExampleData) => void
+  onClick: (exampleData: RenderableExample, isSugestion: boolean) => void
```
Comment (paulmelnikow) [inline frontend/components/badge-examples.tsx]:
```suggestion
  onClick: (exampleData: RenderableExample, isSuggestion: boolean) => void
```

**Comment by calebcartwright** [comment]
One thing I noticed in the review app is that if I change the format type in the copy/dropdown button on the modal window, that the font color on the button changes to black. 

Steps:
1. open the modal window (I used appveyor)
2. enter the required route params
3. click **Copy badge url** button, all is normal
4. change the format type (for example **Copy Markdown**), color changes

Was this expected?

**Comment by paulmelnikow** [comment]
> Was this expected?

Nope! Will fix. I actually hadn't tested this yet – was just working blindly to satisfy the compiler –which says something about how many bugs full typings prevent!

**Comment by calebcartwright** [comment]
> was just working blindly to satisfy the compiler

😆

**Comment by paulmelnikow** [comment]
Fixed!

**Comment by paulmelnikow** [comment]
Thanks for reviewing!

**Pull request merged**